### ISSUE TITLE
Mutation errors on ~2@dev branch

### ISSUE THREAD

**Issue by swthate** [markhuot/craftql#251]
CraftQL: ~2@dev
Craft: 3.1.18

I'm attempting a pretty straightforward mutation query:

```graphql
mutation newTownNews($title: String!, $townId: [Int]) {
  upsertTownNews(title: $title, town: $townId) {
    id
    url
  }
}
```

And I get the following error:
```
{
  "error": "Trying to get property 'section' of non-object"
}
```

I tried rolling back to CraftQL 1.3.1 and got a different error:
```
{
  "error": "Validation failed.\n\n- Could not generate a unique URI based on the URI format."
}
```

This section does have a complex URI: `towns/{object.town.one.title|kebab}/news/{object.title|kebab}`, so I changed it to a very simple URI: `town-news/{slug}` and the mutation worked!

So after finishing a little troubleshooting on my own end I'm dealing with two different issues:

1. CraftQL 1.3.1 is having a hard time with my complex URI. Is `{object.town.one.title|kebab}` the culprit? The structure of the site relies on that URI pattern so I can't really change that.
2. The CraftQL 2beta seems to just have a hard time in general with what seems to be a simple mutation. It might just be failing less gracefully as a result of the same URI issue, though. I'm not sure.

**Comment by swthate** [comment]
Has anyone else here experienced any similar issues when attempting basic mutations in 2.0.0 beta?

**Comment by monachilada** [comment]
Yes, I'm seeing this too. I can update fine when providing the id of an existing entry, but trying to add a new entry returns the same result. I guess it's because the insert is failing and so then when the attempt to return results happens it finds nothing and throws this message. So what is actually going on is upserting is failing for some (validation?) reason.
